baselines/dom.generated.d.ts

```diff
@@ -11834,20 +11864,70 @@ declare var Range: {
     readonly START_TO_START: number;
 };
 
-interface ReadableStream {
+interface ReadableByteStreamController {
+    readonly byobRequest: ReadableStreamBYOBRequest | undefined;
+    readonly desiredSize: number | null;
+    close(): void;
+    enqueue(chunk: ArrayBufferView): void;
+    error(error?: any): void;
+}
+
+interface ReadableStream<R = any> {
     readonly locked: boolean;
-    cancel(): Promise<void>;
-    getReader(): ReadableStreamReader;
+    cancel(reason?: any): Promise<void>;
+    getReader(options: { mode: "byob" }): ReadableStreamBYOBReader;
+    getReader(): ReadableStreamDefaultReader<R>;
+    pipeThrough<T extends ReadableStream>({ writable, readable }: { writable: WritableStream<R>, readable: T }, options?: PipeOptions): T;
```

Comment:
My only niggle is that now that the `WritableReadablePair` interface is gone and with it the generic types that extend _both_ R and W streams, that this is now inconsistent with the rest of the interfaces. It is now the _only_ location that allows for a theoretical type T that extends ReadableStream, versus being a ReadableStream parameterised by a new chunk generic type (O). Like so:

```suggestion
    pipeThrough<O>({ writable, readable }: { writable: WritableStream<R>, readable: ReadableStream<O> }, options?: PipeOptions): ReadableStream<O>;
```
I realise that `T` was also used to not have to have type RS<O> twice, but personally I find this more explicit (and thus clear) and also more consistent with the other interfaces. You can choose any name for type `O` here, I used it to be consistent with output chunk types from TransformStream, but it does differ from `R` but maybe that's fine. This function is one of those instance methods that should have been a free function, but I digress.

Reply:
Fair enough, I guess it's a bit overkill to preserve the subclass type when it's unlikely that people will encounter a subclass of `ReadableStream` in practice.